streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -40,6 +41,9 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> List<T> stores(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        if (streamThread.state() == StreamThread.State.DEAD) {
```

Comment:
I'm wondering why we want to treat `Dead` with `PARTITIONS_REVOKED | etc` differently? I think in both cases we can throw an exception to let the caller re-discover the state stores since they should be migrated to other threads right?